pkg/credentials/credentials.go

```diff
@@ -234,6 +228,73 @@ func (f *fetcher) parseAWSCredentials() (map[string]string, error) {
 	return creds, nil
 }
 
+func openstackAuthOptionsToCreds(opts *gophercloud.AuthOptions, region string) (map[string]string, error) {
+	creds := make(map[string]string)
+	// We could implement some fallbacks here to support e.g. application credentials / domain ID
+	creds[OpenStackAuthURL] = opts.IdentityEndpoint
+	creds[OpenStackUserNameMC] = opts.Username
+	creds[OpenStackPassword] = opts.Password
+	creds[OpenStackDomainName] = opts.DomainName
+	creds[OpenStackRegionName] = region
+
+	if opts.TenantID != "" {
+		creds[OpenStackTenantID] = opts.TenantID
+	}
+
+	if v, ok := creds[OpenStackTenantID]; !ok || len(v) == 0 {
+		creds[OpenStackTenantName] = opts.TenantName
+	}
+
+	return creds, nil
+}
+
+func (f *fetcher) parseOpenStackCredentials() (map[string]string, error) {
+	// 1.) Credentials file loaded
+	if f.Source != nil {
+		return map[string]string{
+			OpenStackAuthURL:    f.F(OpenStackAuthURL),
+			OpenStackPassword:   f.F(OpenStackPassword),
+			OpenStackDomainName: f.F(OpenStackDomainName),
+			OpenStackRegionName: f.F(OpenStackRegionName),
+			OpenStackTenantID:   f.F(OpenStackTenantID),
+			OpenStackTenantName: f.F(OpenStackTenantName),
+			OpenStackUserNameMC: f.F(OpenStackUserName),
+		}, nil
+	}
+
+	// 2.) Check environment variables only use them if all required parameters are set
+	opts, err := openstack.AuthOptionsFromEnv()
+	if err == nil {
+		return openstackAuthOptionsToCreds(&opts, os.Getenv("OS_REGION_NAME"))
+	}
```

Comment:
There are a couple of things I don't like about this approach. I've investigated this function and found out that:

* It fetches all `OS_*` variables. This seems a little bit wrong if we use only a few of them
* It validates `OS_*` variables that we don't use, for example https://github.com/gophercloud/gophercloud/blob/c8c96f74de7f0c91297732d1a1468bc225a68137/openstack/auth_env.go#L99-L110
  * This also seems a little bit wrong because there's no point in validating and blocking on variables that we don't use.
* Their validation is not strict as ours

There are (at least) two potential approaches to this:

* Acknowledge this and go ahead with his approach. The problems with this approach are outlined above
* Continue using `os.Getenv` for the environment variables and use library only for `clouds.yml`

In the second case, we might run into inconsistencies between how environment variables and `clouds.yml` are handled.

Therefore, I'd probably prefer using the approach you proposed in this PR, but we are currently having some internal discussions about this, as well as, we want to test it a little bit.